### PR TITLE
Integrated libocpp patch 

### PR DESCRIPTION
**Pull request by louisg1337**
## Describe your changes

This PR integrates the `libocpp.patch` file that is currently being using in the `charin-e2e-demo`. The goal is to completely get rid of all of the patches so that the `charin-e2e-demo` can be fully functional and not rely on hacky patches. 

**Testing Done**

This code was tested alongside all of the other patches in `everest-demo/charin-e2e-demo`. In the demo, `setChargingProfile`, which this patch helps implement, seems to work on a basic level.

## Issue ticket number and link

The issue that this is related to can be found [here](https://github.com/EVerest/everest-demo/issues/61).

## Checklist before requesting a review
- [x] I have performed a self-review of my code
- [ ] I have made corresponding changes to the documentation
- [ ] If OCPP 2.0.1: I have updated the [OCPP 2.0.1 status document](https://github.com/EVerest/libocpp/tree/main/doc/ocpp_201_status.md)
- [x] I read the [contribution documentation](https://github.com/EVerest/EVerest/blob/main/CONTRIBUTING.md) and made sure that my changes meet its requirements

### CODE REVIEW

**Review by shankari — CHANGES_REQUESTED** [review]
@louisg1337 we should clean up this patch a bit before having others review it. Concretely, we should:
- downgrade all the `EVLOG_error`s to `EVLOG_warning` or `EVLOG_info` or...
- remove out unnecessary commented out code
- make the `MessageType::SetChargingProfile` handling be similar to the others

A lot of this was hacking the code together to figure out what was going on, we should clean it up before submission

File: lib/ocpp/v201/charge_point.cpp [hunk 1]
```diff
@@ -17,6 +17,7 @@
 #include <string>
 
 using namespace std::chrono_literals;
+using namespace nlohmann;
```
Comment (shankari) [inline lib/ocpp/v201/charge_point.cpp]: I don't think we need this.

**Review by shankari — CHANGES_REQUESTED** [review]

**Review by shankari — APPROVED** [review]
@drmrd @couryrr-afs @wjmp please let us know if AFS is able to and wants to review this. Otherwise, we will just merge!

**Comment by ghost** [comment]
This looks fine to me for the demo code.

**Comment by shankari** [comment]
Note that we are requesting to merge this into the main libocpp, not into the demo code.
We currently have this only in the demo code  through patches, but that is not a viable long-term solution.
Also the code to handle the incoming `SetChargingProfile` method is required to hook up the CSMS to the station, so it is most assuredly not demo code. Or are you saying that it will be reimplemented by AFS later?

**Comment by shankari** [comment]
I am going to merge this in the next few days

**Comment by ghost** [comment]
> Note that we are requesting to merge this into the main libocpp, not into the demo code. We currently have this only in the demo code through patches, but that is not a viable long-term solution. Also the code to handle the incoming `SetChargingProfile` method is required to hook up the CSMS to the station, so it is most assuredly not demo code. Or are you saying that it will be reimplemented by AFS later?

@shankari we are actively working on this implementation with test coverage.

**Comment by shankari** [comment]
ok I will merge this change, we can submit to the main everest repo once the tests are done!

**Comment by shankari** [comment]
@louisg1337 can you please change this to point to the `charin-e2e-demo-with-revisions` branch? That is the one that is used in the demo

https://github.com/US-JOET/everest-core/blob/60b4062862460fa7c48a2d79c32c20106d8b8e4a/dependencies.yaml#L59

**Comment by louisg1337** [comment]
> @louisg1337 can you please change this to point to the `charin-e2e-demo-with-revisions` branch? That is the one that is used in the demo

@shankari new PR which points to `charin-e2e-demo-with-revisions` can be found [here](https://github.com/US-JOET/libocpp/pull/31).

**Pull request merged**